### PR TITLE
Adding Hugging Face BERT Notebook

### PR DESCRIPTION
**Pull request by svenchilton**
# Description

Adding a tutorial notebook on optimizing a Hugging Face BERT transformer (specifically for masked language modeling) with Torch-TensorRT

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by svenchilton** [comment]
I'm not sure why the C++ linting is failing. I didn't touch any C++ code. Maybe some C++ files in my branch are out of date with the master branch?

**Review by peri044 — APPROVED** [review]
1) You can probably add the outputs of trt_model_fp16 as well 
2) can you add an explanation why traced model is accepting 3 inputs while TRT compilation requires only 2 inputs ? 
3) Can you modify 5. Benchmarking -> 5. Benchmarking on A100 GPU ? 
Rest LGTM

File: notebooks/Hugging-Face-BERT.ipynb [hunk 1]
```diff
@@ -0,0 +1,972 @@
...
+      "- Remove the dtype spec for attention_mask.1\n",
+      "- Disable partial compilation by setting require_full_compilation to True\n",
+      "WARNING: [Torch-TensorRT] - For input token_type_ids, found user specified input dtype as Int32, however when inspecting the graph, the input type expected was inferred to be Float\n",
+      "The compiler is going to use the user setting Int32\n",
+      "This conflict may cause an error at runtime due to partial compilation being enabled and therefore\n",
+      "compatibility with PyTorch's data type convention is required.\n",
+      "If you do indeed see errors at runtime either:\n",
+      "- Remove the dtype spec for token_type_ids\n",
+      "- Disable partial compilation by setting require_full_compilation to True\n",
+      "WARNING: [Torch-TensorRT TorchScript Conversion Context] - Tensor DataType is determined at build time for tensors not marked as input or output.\n",
```
Comment (narendasan) [inline notebooks/Hugging-Face-BERT.ipynb:356]: Do you think its worth setting the logging level to Error so that these warnings don't get dumped?

Comment (svenchilton) [reply notebooks/Hugging-Face-BERT.ipynb:356]: Sure, that sounds fine. How do I change the logging level?

**Comment by svenchilton** [comment]
@peri044: 

> 1. You can probably add the outputs of trt_model_fp16 as well
For reasons I still don't fully understand (something to do with different precision, but beyond that, I couldn't tell you), I never managed to get the correct outputs when I passed the masked sentences into the FP16 TRT model.

> 2. can you add an explanation why traced model is accepting 3 inputs while TRT compilation requires only 2 inputs ?
I changed the TRT compilation so that the model takes 3 inputs rather than the 2 that you and @narendasan used in your original scripts.

> 3. Can you modify 5. Benchmarking -> 5. Benchmarking on A100 GPU ?
I can certainly do that, but the code will be the same regardless of the user's GPU, right? Might it make more sense to add a note saying that the results shown in GitHub were obtained on an A100 and will vary?

>    Rest LGTM
Yay!

**Comment by narendasan** [comment]
Also @svenchilton can you sign you commits? https://github.com/NVIDIA/Torch-TensorRT/pull/978/checks?check_run_id=6042619063

**Comment by svenchilton** [comment]
I messed up the signoffs, so I'm going to start a new branch with the current notebook and submit a new pull request for that.
